### ISSUE TITLE
Make UI tests agnostic of outside content

### ISSUE THREAD

**Issue by filmaj** [adobe/developer.adobe.com#142]
Some of the current UI tests inspect and assert on content (on homepage and other doc strains). Some of this content is not revisioned in this repo, and so could change at any time. While homepage content is revisioned in this repo, docs content is not.

It would be better to assert on other things for e.g. docs and marketing page strains. Maybe better to assert on template parameters for these pages rather than content?

**Comment by kptdobe** [comment]
Actually, that's the exact reason why I did not use a proper browser framework in my initial smoke tests: I only tested if some dom elements are in the response which does not require anything else that a GET function ;)
It is still not clear to me how we can make those e2e content agnostic.
If you know the content, you know the output, thus fixing the content should certainly be the right solution. Maybe the "code" repo could have a `test/content` with "stable test content" to request against. Not sure if this works. To be tested.

**Comment by filmaj** [comment]
> It is still not clear to me how we can make those e2e content agnostic.

The templates the content is shown in are revisioned in this repo, so we could assert on that.

> If you know the content, you know the output, thus fixing the content should certainly be the right solution. Maybe the "code" repo could have a `test/content` with "stable test content" to request against. Not sure if this works. To be tested.

This is more of a helix discussion (adobe/helix-cli#582) and less of a dev-site discussion, but why not simplify with: have helix pull in the actual content markdown used in the content repo, and as part of whatever CI/CD framework helix recommends, ensure that content is present SOMEWHERE in the final URL?

For example, given the following `helix-config.yaml` section:

```
  xd-docs-helixdemo:
    <<: *basestrain
    url: https://adobedevsite.helix-demo.xyz/xd/docs
    content: https://github.com/AdobeXD/plugin-docs.git#master
    directoryIndex: README.html
    package: 77017e262a9aa0378abd02c098ecc82e9b6aa7dc
```

Helix would:

1. Grab the content from https://github.com/AdobeXD/plugin-docs.git/blob/master/README.md, from the master branch.
2. `curl` (or use whatever tool) to get the contents of https://adobedevsite.helix-demo.xyz/xd/docs
3. Somehow ensure some (all?) content from (1) is present in the response from (2).

**Comment by kptdobe** [comment]
You are right saying it is an Helix discussion topic. But since you have a real life example... ;)

Your solution does not work because the dev (via the `pre.js` and `htl` template) can do whatever he wants with the content: do not render it, transform it... there is no way to guess the result.

Only a dev would know for input A, I must have output B and tests must be written with certitudes, not guesses.
